extension/app/ts/background/windows/personalSign.ts

```diff
@@ -43,85 +46,124 @@ function reject() {
-	if (params.method === 'personal_sign') {
+export async function craftPersonalSignPopupMessage(ethereumClientService: EthereumClientService, originalParams: PersonalSignParams | SignTypedDataParams | OldSignTypedDataParams, activeAddress: bigint, userAddressBook: UserAddressBook, simulationMode: boolean, requestId: number, signerName: SignerName, website: Website): Promise<PersonalSignRequest> {
+	const activeAddressWithMetadata = getAddressMetaData(activeAddress, userAddressBook)
+	const basicParams = {
+		activeAddress: activeAddressWithMetadata,
+		simulationMode,
+		requestId,
+		website,
+		signerName,
+	}
+
+	const getQuarrantineCodes = async (messageChainId: bigint, account: AddressBookEntry, activeAddress: AddressBookEntry, owner: AddressBookEntry | undefined): Promise<{ quarantine: boolean, quarantineCodes: readonly QUARANTINE_CODE[] }> => {
+		let quarantineCodes: QUARANTINE_CODE[] = []
+		if (BigInt(messageChainId) !== (await getSettings()).activeChain) {
+			quarantineCodes.push('SIGNATURE_CHAIN_ID_DOES_NOT_MATCH')
+		}
+		if (account.address !== activeAddress.address || (owner != undefined && account.address !== owner.address)) {
+			quarantineCodes.push('SIGNATURE_ACCOUNT_DOES_NOT_MATCH')
+		}
+		return {
+			quarantine: quarantineCodes.length > 0,
+			quarantineCodes,
+		}
+	}
+	if (originalParams.method === 'eth_signTypedData') {
 		return {
 			method: 'popup_personal_sign_request',
 			data: {
-				activeAddress,
+				originalParams,
+				...basicParams,
 				type: 'NotParsed' as const,
-				simulationMode,
-				requestId,
-				message: params.params[0],
-				account: getAddressMetaData(params.params[1], userAddressBook),
-				method: params.method,
-				params,
+				message: stringifyJSONWithBigInts(originalParams.params[0]),
+				account: getAddressMetaData(originalParams.params[1], userAddressBook),
+				quarantine: false,
+				quarantineCodes: [],
 			}
 		} as const
 	}
 
-	if (params.params[1].primaryType === 'Permit') {
-		const parsed = EIP2612Message.parse(params.params[1])
+	if (originalParams.method === 'personal_sign') {
 		return {
 			method: 'popup_personal_sign_request',
 			data: {
-				activeAddress,
+				originalParams,
+				...basicParams,
+				type: 'NotParsed' as const,
+				message: originalParams.params[0],
+				account: getAddressMetaData(originalParams.params[1], userAddressBook),
+				quarantine: false,
+				quarantineCodes: [],
+			}
+		} as const
+	}
+	const namedParams = { param: originalParams.params[1], account: originalParams.params[0] }
+	const account = getAddressMetaData(namedParams.account, userAddressBook)
+
+	if (namedParams.param.primaryType === 'Permit') {
+		const parsed = EIP2612Message.parse(namedParams.param)
+		const token = await getTokenMetadata(ethereumClientService, parsed.domain.verifyingContract)
+		const owner = getAddressMetaData(parsed.message.owner, userAddressBook)
+		if (token.type === 'NFT') throw 'Attempted to perform Permit to an NFT'
```

Comment:
Does permit not work with NFTs?  Is there some other standard for permitting NFTs?

Reply:
permit and permit2 are only for erc20: https://eips.ethereum.org/EIPS/eip-2612. I guess this is the same for NFT's but its a draft: https://eips.ethereum.org/EIPS/eip-4494

Reply:
I added issue for eip4494, I didn't directly find a dapp that would use it tho